gqcp/include/ONVBasis/SpinResolvedOperatorString.hpp

```diff
@@ -0,0 +1,169 @@
+
+    /**
+     * Return the spin resolved operator string split in its two separate components, one containing the alpha operators, one containing the beta operators.
+     */
+    SpinResolved<SpinUnresolvedOperatorString> SpinResolve() const {
+        // Split pairs in two separate spinUnresolvedOPeratorStrings.
+        // Define a separate vector for the alpha and beta components.
+        auto alpha_index_vector = std::vector<size_t> {};
+        auto beta_index_vector = std::vector<size_t> {};
+
+        // Loop over the original pair vector. If the pair contains an alpha spin, add the index to the alpha vector, if the pair contains a beta spin, add the index to the beta vector.
+        for (int i = 0; i < this->index_spin_pairs.size(); i++) {
+            if (this->index_spin_pairs[i].second == GQCP::Spin::alpha) {
+                alpha_index_vector.push_back(index_spin_pairs[i].first);
+            } else {
+                beta_index_vector.push_back(index_spin_pairs[i].first);
+            }
+        }
+
+        // Determine the phase factor needed to make the splitting of the original operator string in its spin components correspond with the fermion anti-commutation rules.
+        // We will define an intermediate phase factor, which is updated throughout the procedure.
+        int intermediate_phase_factor = 1;
+
+        // First, we loop the spin string to check whether an element has alpha spin.
+        for (int i = 0; i < this->operatorSpins().size(); i++) {
+
+            // If it does, we continue, if it doesn't, we move on to the next pair in the vector.
+            if (this->operatorSpins()[i] == GQCP::Spin::alpha) {
+
+                // We check all elements left of our found alpha element, to see whether or not a beta element is found.
+                for (int j = i; j >= 0; j--) {
+
+                    // If we find a beta element left of the alpha element, we must swap the two elements in the vector and update the phase factor by multiplying it by -1, due to the anti-commutation rules.
+                    // We don't physically swap the elements in the vector, but by simply checking the number of beta's left of the alpha in question, we mimic the swap.
+                    if (this->operatorSpins()[j] == GQCP::Spin::beta) {
+                        intermediate_phase_factor *= -1;
+                    }
+                }
+            }
+        }
+
+        // The phase factor has been updated in the loops, in order to correspond to a modified operator string with all alpha operators first, followed by all beta operators.
+        // The total operator string is applied on |vac_a>|vac_b>, e.g.: a_a a_a a_b a_b|vac_a>|vac_b>. But we want a_a a_a |vac_a> a_b a_b|vac_b>.
+        // Now we have to apply the fermion anti-commutation rules for eacht time we move |vac_alpha> over a beta operator. This results in p^N_beta.
+        int final_phase_factor = pow(intermediate_phase_factor, beta_index_vector.size());
```

Comment:
Moving the (true) vacuum doesn't affect the phase factor since it's empty. I think the correct implementation is the one that ends at line `157`. Could you verify this?

Reply:
You are correct, this one is on me. In the notes from our meeting we split the string in `D_a` and `I_a` and while implementing I mistook moving `I_a` part over the beta operators for moving the vacuum. 

As it is empty, moving the vacuum indeed doesn't change the phase factor. I'll fix this and adjust the documentation where necessary.

Reply:
@lelemmen I just realized my explanation above wasn't entirely clear. The `I_a` in our notes represented the alpha creation operators acting on the alpha vacuum, which in total represented the ONV on which the operator string should be acting. 

When moving these alpha operators across the beta operators of the spin resolved operator string, we have to take the  anti-commutation rules into account. However, this is context dependant (since it depends on the ONV on which the operator string acts), and should not be factored in in this implementation, as here, we focus purely on the operator string itself. 

Long story short, I did indeed make a mistake here & you are correct that what this API has to do is implemented up until line 157, as moving the vacuum has no effect. However, when using this class in for example the future `calculateNDMElement` API's, we still have to take the anti commutation rules of moving the `I_a^dagger` operators.

Hope this clears things up!

Reply:
Yes, the extra anti-commutation relations resulting from `I_a^dagger` should be accounted for, but (as you said) not in this class.